pkg/skaffold/build/buildpacks/init.go

```diff
@@ -69,7 +71,22 @@ func (c ArtifactConfig) Path() string {
 
 // validate checks if a file is a valid Buildpack configuration.
 func validate(path string) bool {
-	name := filepath.Base(path)
+	switch filepath.Base(path) {
+	case "package.json":
+		return !hasParent(path, "node_modules")
+	case "go.mod":
+		return !hasParent(path, "vendor")
```

Comment:
Seems worth emitting logging something here to inform the user that the project was skipped due to node_modules/vendor.  I know I'd puzzled as to why one "clean" project is picked up both another ignored.  I'm sure there are a number of projects that commit their vendor'd folders (like skaffold).

And a todo to correlate this with the [build.exclude](https://github.com/buildpacks/rfcs/blob/master/text/0019-project-descriptor.md#buildinclude-and-buildexclude)?

Reply:
I'm not sure I understand your point. We are ignoring what's under `vendor` and `node_modules`. Wether those folders exist or not give the same result.

Reply:
:blush: I'm sorry, I did not dig into this code hard enough.  From our previous conversations about interference with buildpacks from projects with `node_modules` or `vendor`, the name `validate` threw me off: I thought this code was to identify problematic projects with an accompanying `node_modules` or `vendor`.